File: core/src/main/java/com/gmail/marcosav2010/communicator/packet/AbstractPacket.java
package com.gmail.marcosav2010.communicator.packet;

import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;

import java.io.IOException;

public abstract class AbstractPacket {

    public static int BASE_SIZE = Short.MAX_VALUE * 32 * 4; // ~4 MB
    public static int MAX_SIZE = BASE_SIZE - Byte.BYTES;

    public abstract void encode(PacketEncoder out) throws IOException;

    public abstract void decode(PacketDecoder in) throws IOException;

    public abstract boolean isStandard();
}

File: core/src/main/java/com/gmail/marcosav2010/communicator/packet/AbstractPacket.kt
package com.gmail.marcosav2010.communicator.packet

import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder

abstract class AbstractPacket {

    companion object {
        var BASE_SIZE = Short.MAX_VALUE * 32 * 4 // ~4 MB

        @JvmField
        var MAX_SIZE = BASE_SIZE - java.lang.Byte.BYTES
    }

    abstract fun encode(out: PacketEncoder)

    abstract fun decode(`in`: PacketDecoder)

    abstract val isStandard: Boolean
}